### PR TITLE
Improve tooltips

### PR DESCRIPTION
**Pull request by BinderDavid**
Improves the tooltips for AreaCharts.

### CODE REVIEW

File: src/VegaTemplate.hs [hunk 1]
```diff
@@ -196,6 +196,8 @@ encodingBandsLayer ct c =
   encoding
     . order [OName "k", OmType Quantitative]
     . color [MName "c", MmType Nominal, MScale [colourProperty c], MLegend []]
+    . injectJSON "tooltip" (toJSON [object ["field" .= String "x", "type" .= String "quantitative", "title" .= String "Time", "format" .= String ".3f"],
```
Comment (mpickering) [inline src/VegaTemplate.hs]: This should be the `y` field? Allocations?

File: src/VegaTemplate.hs [hunk 2]
```diff
@@ -237,15 +238,8 @@ encodingTracesLayer =
...
-    . opacity [MSelectionCondition (Expr "index") [MNumber 1] [MNumber 0.5]]
     -- The "tooltips" feature is not in the current version of HVega
-    . injectJSON "tooltip" (toJSON [object ["field" .= String "tx", "type" .= String "quantitative"],
-                             object ["field" .= String "desc", "type" .= String "nominal"]])
-
-selectionTracesLayer ::  [LabelledSpec] -> (VLProperty, VLSpec)
-selectionTracesLayer =
-  selection
-    . VL.select "index" Single [On "mousemove", Encodings [ChX], Nearest True]
+    . injectJSON "tooltip" (toJSON [object ["field" .= String "desc", "type" .= String "nominal", "title" .= String "Event"]])
```
Comment (mpickering) [inline src/VegaTemplate.hs]: Title is redundant here?

**Comment by mpickering** [comment]
Do you have a screenshot to hand of how this looks?

File: src/VegaTemplate.hs (hunk 2 above)
Comment (BinderDavid) [reply src/VegaTemplate.hs]: Ok, In that case I can even replace this by the HVega library function.

**Comment by mpickering** [comment]
You need to change the trace tooltip from a list to a single object so it doesn't display the field name.

**Comment by BinderDavid** [comment]
Screenshot from 2019-06-23 17-32-44

**Comment by mpickering** [comment]
Looks good. I tried locally and the CSS framework changes how the tooltips are rendered. Looking into how to fix that now.

**Comment by mpickering** [comment]
I fixed the rendering of tooltips with the stylesheet. I'll make a PR once this is merged.

File: src/VegaTemplate.hs (hunk 1 above)
Comment (mpickering) [reply src/VegaTemplate.hs]: This still needs to be changed?

Comment (BinderDavid) [reply src/VegaTemplate.hs]: What do you want the tooltip to show when hovering over a band?
I currently show the time, which in the Json data is encoded as "x", and the Type, which is encoded as "c".
If we want to show the Allocation as well, I think we have to use not "y" but the name of the field of the underlying data, i.e. "Allocation"

Comment (mpickering) [reply src/VegaTemplate.hs]: I want the hover to show allocation rather than time, so "Allocation" is right.

**Comment by BinderDavid** [comment]
Ok, it was indeed "y". I thought the JSON contained an "Allocation" field.

**Comment by mpickering** [comment]
Performance is much better for me after this PR as an unintended benefit or not using the interactive feature.

**Pull request merged**